### PR TITLE
Launcher update

### PR DESCRIPTION
**Pull request by iamawesomecat**

### CODE REVIEW

File: src/main/java/frc/team2412/robot/commands/launcher/SetLaunchSpeedCommand.java [hunk 1]
```diff
@@ -16,6 +16,7 @@ public SetLaunchSpeedCommand(LauncherSubsystem launcherSubsystem, double speed)
 
 	@Override
 	public void initialize() {
+		System.out.println("Setting Angle!");
```
Comment (TAKBS2412) [inline src/main/java/frc/team2412/robot/commands/launcher/SetLaunchSpeedCommand.java]: Should revert this, we log when commands are initialized [here](https://github.com/robototes/Crescendo2024/blob/798f7938ddbb0e3b074043232ab81e3c1fca8023/src/main/java/frc/team2412/robot/Robot.java#L83).

File: src/main/java/frc/team2412/robot/subsystems/LauncherSubsystem.java [hunk 2]
```diff
@@ -152,6 +152,12 @@ public double getAngle() {
 
 	public void setAngle(double angle) {
 		launcherAnglePIDController.setReference(Units.degreesToRotations(angle), ControlType.kPosition);
+		System.out.println("Setting Angle!");
```
Comment (TAKBS2412) [inline src/main/java/frc/team2412/robot/subsystems/LauncherSubsystem.java]: Same for this print message - we should also delete.

File: src/main/java/frc/team2412/robot/commands/launcher/SetAngleCommand.java [hunk 3]
```diff
@@ -17,6 +17,8 @@ public SetAngleCommand(LauncherSubsystem launcherSubsystem, double angle) {
 
 	@Override
 	public void initialize() {
+		System.out.println("Setting Angle!");
```
Comment (TAKBS2412) [inline src/main/java/frc/team2412/robot/commands/launcher/SetAngleCommand.java]: Should revert this, we log when commands are initialized [here](https://github.com/robototes/Crescendo2024/blob/798f7938ddbb0e3b074043232ab81e3c1fca8023/src/main/java/frc/team2412/robot/Robot.java#L83).

File: src/main/java/frc/team2412/robot/subsystems/LauncherSubsystem.java [hunk 4]
```diff
@@ -152,6 +152,12 @@ public double getAngle() {
...
+	}
+
+	public boolean isAtTargetSpeed() {
+		return (setLauncherSpeedEntry.getDouble(0)
+				== launcherSpeedEntry.getDouble(SPEAKER_SHOOT_SPEED_RPM));
```
Comment (TAKBS2412) [inline src/main/java/frc/team2412/robot/subsystems/LauncherSubsystem.java]: Would prefer to get the velocity directly from the encoder, instead of relying on `launcherSpeedEntry` to always be updated.

Also, if we are using `setLauncherSpeedEntry` to store the target speed, we should update it in `launch(speed)`, otherwise this will return false when we use that method.

File: src/main/java/frc/team2412/robot/commands/launcher/LaunchCommand.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.team2412.robot.subsystems.IntakeSubsystem;
+import frc.team2412.robot.subsystems.LauncherSubsystem;
+
+public class LaunchCommand extends Command {
+	private final LauncherSubsystem launcherSubsystem;
+	private final IntakeSubsystem intakeSubsystem;
+
+	public LaunchCommand(LauncherSubsystem launcherSubsystem, IntakeSubsystem IntakeSubsystem) {
```
Comment (TAKBS2412) [inline src/main/java/frc/team2412/robot/commands/launcher/LaunchCommand.java]: nit: make the `IntakeSubsystem` variable camelcase.

File: src/main/java/frc/team2412/robot/Subsystems.java [hunk 6]
```diff
@@ -14,7 +14,7 @@ public static class SubsystemConstants {
 		public static final boolean APRILTAGS_ENABLED = false;
 		public static final boolean LIMELIGHT_ENABLED = false;
 		public static final boolean CLIMB_ENABLED = false;
-		public static final boolean LAUNCHER_ENABLED = false;
+		public static final boolean LAUNCHER_ENABLED = true;
```
Comment (TAKBS2412) [inline src/main/java/frc/team2412/robot/Subsystems.java]: Should revert, typically we do not enable subsystems until they're physically ready on the robot.

File: src/main/java/frc/team2412/robot/Controls.java [hunk 7]
```diff
@@ -95,15 +98,19 @@ private void bindIntakeControls() {
...
+		// launcherPodiumPresetButton.onTrue(
+		//		new SetAngleLaunchCommand(
+		//				s.launcherSubsystem,
+		//				LauncherSubsystem.SPEAKER_SHOOT_SPEED_RPM,
+		//				LauncherSubsystem.PODIUM_AIM_ANGLE));
+		// launcherSubwooferPresetButton.onTrue(
+		//		new SetAngleLaunchCommand(
+		//				s.launcherSubsystem,
+		//				LauncherSubsystem.SPEAKER_SHOOT_SPEED_RPM,
+		//				LauncherSubsystem.SUBWOOFER_AIM_ANGLE));
```
Comment (TAKBS2412) [inline src/main/java/frc/team2412/robot/Controls.java:117]: Curious, why were these deleted? (now we have no way to use presets).

**Review by TAKBS2412 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/team2412/robot/subsystems/LauncherSubsystem.java (hunk 2 above)
Comment (iamawesomecat) [reply src/main/java/frc/team2412/robot/subsystems/LauncherSubsystem.java]: done

File: src/main/java/frc/team2412/robot/commands/launcher/SetAngleCommand.java (hunk 3 above)
Comment (iamawesomecat) [reply src/main/java/frc/team2412/robot/commands/launcher/SetAngleCommand.java]: done

File: src/main/java/frc/team2412/robot/commands/launcher/SetLaunchSpeedCommand.java (hunk 1 above)
Comment (iamawesomecat) [reply src/main/java/frc/team2412/robot/commands/launcher/SetLaunchSpeedCommand.java]: done

File: src/main/java/frc/team2412/robot/commands/launcher/LaunchCommand.java (hunk 5 above)
Comment (iamawesomecat) [reply src/main/java/frc/team2412/robot/commands/launcher/LaunchCommand.java]: done

File: src/main/java/frc/team2412/robot/Subsystems.java (hunk 6 above)
Comment (iamawesomecat) [reply src/main/java/frc/team2412/robot/Subsystems.java]: done, dw

File: src/main/java/frc/team2412/robot/Controls.java (hunk 7 above)
Comment (iamawesomecat) [reply src/main/java/frc/team2412/robot/Controls.java:117]: presets, deleted for now as intake is using the keybinds we needed, it will be re-instated if we need presets right before comp or if the keybinding thing is resolved.

File: src/main/java/frc/team2412/robot/subsystems/LauncherSubsystem.java (hunk 4 above)
Comment (iamawesomecat) [reply src/main/java/frc/team2412/robot/subsystems/LauncherSubsystem.java]: 1. bet


2.launch(speed) is mainly for presets, if they set a speed before a match, then it'll get messed up during auto.

**Review by jamesdooley4 — APPROVED** [review]

**Review by TAKBS2412 — APPROVED** [review]
